src/web3/token-vote/graphql/graphql.service.ts

```diff
@@ -49,4 +52,28 @@ export class GraphqlService {
         return result;
     }
 
+    async getEquivalentBlock(blockHeight, chainId) {
```

Comment:
I wonder if we need safeguards for out of bounds. Ethereum block 1 existed before Polygon ever did, so in this case Polygon would be excluded from the strategy completely. 

Might need to log it for later.

Reply:
There is a safeguard on line 71. I don't remeber where the number comes from though probably needs double checking. Also we're not handling that case properly. I'll try to fix that along with some retry logic on rate limit